### PR TITLE
OEUI-131: Use a constant for the date format

### PR DESCRIPTION
**Pull request by zeze111**
[OEUI-131: Use a constant for the date format](https://issues.openmrs.org/browse/OEUI-131)

### SUMMARY: 
The date format is used in two files, however, for maintainability, the date format is gotten from the global properties that has been set online. This PR creates actions to call the API endpoint to get the date format global property, and then stores it in the redux-store, which is then easily accessible by the required components.

### CODE REVIEW

File: __mocks__/mockData.js [hunk 1]
```diff
@@ -191,4 +191,3 @@ export default {
   itemName: '',
   attributes: [],
 };
-
```
[the remark this answers is not in the document]
Comment (zeze111) [reply __mocks__/mockData.js]: I didn't, the line is there, I'm confused why it came up

[the remark this answers is not in the document]
Comment (fuchodeveloper) [reply __mocks__/mockData.js]: Looks like you deleted the last line. 
You could add a return there to fix this.

File: app/js/components/orderEntry/OrderEntryPage.jsx [hunk 2]
```diff
@@ -78,6 +85,32 @@ export class OrderEntryPage extends React.Component {
...
+            {
+              dateError === 'incomplete config' ?
+                <span>
+                  please ensure that you have created a valid <strong>date and time format</strong>.
+                </span> :
+                <span>
+                  ensure that you have created a setting called
+                  <strong>orderentryowa.dateAndTimeFormat</strong>
+                  &nbsp;
+                  with a value corresponding to DD-MM-YYYY HH:mm.
```
Comment (dkayiwa) [inline app/js/components/orderEntry/OrderEntryPage.jsx]: Why do you hard code the above date format?

Comment (zeze111) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: It's not reflective as a date, it's a sample of what the value is supposed to be, when the user is setting the date and time format in the global properties

Comment (dkayiwa) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: Then the message needs to change and reflect so.

Comment (zeze111) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: @dkayiwa I've made changes, please take a look

File: README.md [hunk 3]
```diff
@@ -162,11 +162,13 @@ Sharing in Tomcat.
 See instructions [here](http://enable-cors.org/server_tomcat.html) for Tomcat 7 and [here](https://www.dforge.net/2013/09/16/enabling-cors-on-apache-tomcat-6/) for Tomcat 6.
 
 ## Usage
-Before using the Order Entry Open Web Application, the administrator should make the following one time configurations using the admin portal. 
+Before using the Order Entry Open Web Application, the administrator should make the following one time configurations using the admin portal.
 1. Create an Encounter Type called `Order Entry` or any other name.
 2. Create a setting `order.encounterType` whose value should correspond to the name given to the Encounter Type created above.
 3. Create an Encounter Role called `Clinician` or any other name.
 4. Create a setting `order.encounterRole` whose value should correspond to the name given to the encounter role created in 3 above.
+5. Create a date format, setting `orderentryowa.dateAndTimeFormat` as the name, with a value of `DD-MMM-YYYY HH:mm`, in the global properties
```
Comment (dkayiwa) [inline README.md]: How about here?

Comment (zeze111) [reply README.md]: @dkayiwa done

File: README.md [hunk 4]
```diff
@@ -162,11 +162,13 @@ Sharing in Tomcat.
 See instructions [here](http://enable-cors.org/server_tomcat.html) for Tomcat 7 and [here](https://www.dforge.net/2013/09/16/enabling-cors-on-apache-tomcat-6/) for Tomcat 6.
 
 ## Usage
-Before using the Order Entry Open Web Application, the administrator should make the following one time configurations using the admin portal. 
+Before using the Order Entry Open Web Application, the administrator should make the following one time configurations using the admin portal.
 1. Create an Encounter Type called `Order Entry` or any other name.
 2. Create a setting `order.encounterType` whose value should correspond to the name given to the Encounter Type created above.
 3. Create an Encounter Role called `Clinician` or any other name.
 4. Create a setting `order.encounterRole` whose value should correspond to the name given to the encounter role created in 3 above.
+5. Create a date format, setting `orderentryowa.dateAndTimeFormat` as the name, with a sample value of `DD-MMM-YYYY HH:mm`, in the global properties
```
Comment (dkayiwa) [inline README.md]: This is not as clear as the one you made below.

Comment (zeze111) [reply README.md]: @dkayiwa I've corrected it

**Pull request merged**